e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
Because the approach that has been discussed is to do this:
````
assert(x);
if (!x)
    do_error_handling();
````
This provides us with an abort() in a dev build and a properly handled error in a prod build. That's fine but it necessarily requires us to write out the whole expression "x" twice. That is just going to be a source of errors in itself, i.e. discrepancies between the first and second version (especially since one is a negation of the other). This approach only requires us to have one version of x. So, instead your write:
````
if(!ossl_assert(x))
    do_error_handling();
````

Or perhaps you meant "what is wrong with plain old assert() instead of using `if (!expr) OPENSSL_die()`?". If so, that doesn't work so well because assert() is a macro that prints the contained expression on the screen before aborting. If we used it in place of OPENSSL_die() the expression it printed would always be "expr", which isn't too helpful.